Alternate Top App Bar for settings pages

byte2pixel opened the pull request:
> [!Note]
> All relevant changes are in HomeScreen.kt.
> The changes to CommonUI.kt were just removing unused imports and doing a "format code"

# Settings Top App Bar

It came to my attention in testing that the settings screens are meant to have their own top app bar.
I created an alternate top app bar for when the user is on a settings page.

## Steps to test

1. On the "Feed" page.
2. Open the menu
3. Click Edit Account
4. You'll see:
<img width="557" height="290" alt="image" src="https://github.com/user-attachments/assets/4c7be5e7-f11d-493a-9629-52bde5bde110" />

5. Click the Back button
6. Click the menu.
7. Click "About"
8. You'll see:
<img width="553" height="382" alt="image" src="https://github.com/user-attachments/assets/31d595c3-ef49-4db0-996c-0171ff82bc88" />

This app bar could be moved to Common UI and used for the "No Account" screen.

mr-wes7694 commented:
I did not realize this was pending review before I did my commit. However I dont think either of our changes conflict. This looks good to me so far, I appreciate the back arrows as well. It looks really good.

byte2pixel commented:
> I did not realize this was pending review before I did my commit. However I dont think either of our changes conflict. This looks good to me so far, I appreciate the back arrows as well. It looks really good.

Nope, no conflicts.  All good.

I did update this PR after your comment.

I merged your changes in after I approved them since everything worked well when I tested it, there were just minor "house keeping" things.  I addressed in my updated commit.

https://github.com/Cos229-239/2025_08_Team2/pull/31/commits/f3f31711039108824d36fc6bf7be5481a0976b5c

I also redid the routing logic to locate it in the navigation package and removed the enum that was in HomeScreen.kt
